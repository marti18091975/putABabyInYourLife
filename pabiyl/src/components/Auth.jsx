import React, { useState } from "react";
import "firebase/auth";
import { useFirebaseApp, useUser } from "reactfire";
export default (props) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const firebase = useFirebaseApp();
  const user = useUser();
  const submit = async () => {
    await firebase.auth().createUserWithEmailAndPassword(email, password);
  };
  const logout = async () => {
    await firebase.auth().signOut();
  };
  const login = async () => {
    await firebase.auth().signInWithEmailAndPassword(email, password);
  };
  const signInWithGoogle = async () => {
    const provider = new firebase.auth.GoogleAuthProvider();
    provider.addScope("https://www.googleapis.com/auth/contacts.readonly");
    await firebase.auth().signInWithPopup(provider);
  };

  const signInAnonnymusly = async () => {
    await firebase.auth().signInAnonymously();
  };
  return (
    <div>
      {!user && (
        <div>
          <label htmlFor="email">E-mail</label>
          <input
            type="email"
            id="email"
            onChange={(ev) => setEmail(ev.target.value)}
          ></input>
          <label htmlFor="password">Password</label>
          <input
            type="password"
            id="password"
            onChange={(ev) => setPassword(ev.target.value)}
          ></input>
          <button onClick={submit}>register</button>
          <button onClick={login}>Start session</button>
          <button onClick={signInWithGoogle}>Google</button>
          <button onClick={signInAnonnymusly}>Anon</button>
        </div>
      )}
      {user && <button onClick={logout}>Log out</button>}
    </div>
  );
};
