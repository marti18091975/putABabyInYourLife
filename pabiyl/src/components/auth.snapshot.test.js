import renderer from 'react-test-renderer';
import React from 'react';
import Auth from './Auth';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {

    const tree = renderer.create(
        <Router>
            <Auth />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});