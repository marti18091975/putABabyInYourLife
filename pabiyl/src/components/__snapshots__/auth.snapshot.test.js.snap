// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<div
  className="main__login"
>
  <div
    className="login-container"
  >
    <img
      alt=""
      className="logo__login"
      src="https://www.flaticon.es/svg/static/icons/svg/2444/2444603.svg"
    />
    <label
      htmlFor="email"
    >
      E-mail
    </label>
    <input
      className="email__box"
      onChange={[Function]}
      type="email"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="email__box"
      id="password"
      onChange={[Function]}
      pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
      required={true}
      title="A valid password must have more than 7 characters, an uppercase letter, a lowercase letter and a number"
      type="password"
    />
    <div
      className="section__bottom"
    >
      <div
        className="section"
      >
        <button
          className="button__login login"
          onClick={[Function]}
        />
        <button
          className="button__login register"
          onClick={[Function]}
        />
      </div>
      <div
        className="section"
      >
        <button
          className="button__login google"
          onClick={[Function]}
        />
        <button
          className="button__login anon"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
